pkg/channel/types.go

```diff
@@ -49,5 +49,32 @@ const (
 
 // String represent of Type enum
 func (t Type) String() string {
-	return [...]string{"LISTENER", "SENDER", "EVENT" ,"STATUS"}[t]
+	return [...]string{"LISTENER", "SENDER", "EVENT", "STATUS"}[t]
+}
+
+// EventDataType types passed to cloud events
+type EventDataType string
+
+const (
+	// PTPEvent event type is ptp
+	PTPEvent EventDataType = "PTP_EVENT"
+	// PTPStatus event type is ptp status
+	PTPStatus = "PTP_STATUS"
+	// HWEvent event type is hw
+	HWEvent = "HW_EVENT"
+	// Event event type is generic event
+	Event = "EVENT"
+)
+
+// String return string eventType
+func (e EventDataType) String() string {
+	EventType := [...]string{"PTP_EVENT", "PTP_STATUS", "HW_EVENT", "EVENT"}
+
+	x := string(e)
+	for _, v := range EventType {
+		if v == x {
+			return x
+		}
+	}
+	return "EVENT"
```

Comment:
should we return `nil` or `""` here since it would be hard to debug when any typos are resulting in `EVENT`.

Reply:
by default it has to be EVENT type, shouldn't be an empty data type.
Either EVENT or any other Type. Right now we are checking only for HWEvent  data type only rest  goes to default